device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/IotHubConnectionStateCallback.java

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.azure.sdk.iot.device;
+
+/**
+ * An interface for an IoT Hub connection state callback.
+ *
+ * Developers are expected to create an implementation of this interface,
+ * and the transport will call {@link IotHubConnectionStateCallback#connectionUp()}
+ * when connection is established with the IoT Hub and call
+ * {@link IotHubConnectionStateCallback#connectionDown()} when connection is lost.
+ */
+public interface IotHubConnectionStateCallback {
+    /**
+     * Method executed when the connection with the IoTHub is lost.
+     */
+    void connectionUp();
```

Comment:
I will add the enum and also change the callback up to only have an execute method that takes in the enum and a context object.